src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeKernelMap.cs

```diff
@@ -130,9 +130,9 @@ protected virtual void Dispose(bool disposing)
             int radius = (int)TolerantMath.Ceiling(scale * sampler.Radius);
 
             // 'ratio' is a rational number.
-            // Multiplying it by LCM(sourceSize, destSize)/sourceSize will result in a whole number "again".
+            // Multiplying it by destSize/GCD(sourceSize, destSize) will result in a whole number "again".
             // This value is determining the length of the periods in repeating kernel map rows.
-            int period = Numerics.LeastCommonMultiple(sourceSize, destinationSize) / sourceSize;
+            int period = destinationSize / Numerics.GreatestCommonDivisor(sourceSize, destinationSize);
```

Comment:
Cannot place a comment on line `122` but do we need to protect this method from division by zero?

Reply:
GCD should always yield a positive integer unless both inputs are 0. That can never happen in our code since the `ResizeHelper` will always return positive integers.